Sources/Build/BuildDelegate.swift

```diff
@@ -171,31 +188,34 @@ extension SPMLLBuild.Diagnostic: DiagnosticDataConvertible {
 }
 
 private let newLineByte: UInt8 = 10
-public final class BuildDelegate: BuildSystemDelegate {
-    // Track counts of commands based on their CommandStatusKind
-    private struct CommandCounter {
-        var scanningCount = 0
-        var upToDateCount = 0
-        var completedCount = 0
-        var startedCount = 0
-
-        var estimatedMaximum: Int {
-            return completedCount + scanningCount - upToDateCount
-        }
-
-        mutating func update(command: SPMLLBuild.Command, kind: CommandStatusKind) {
+public final class BuildDelegate: BuildSystemDelegate, SwiftCompilerOutputParserDelegate {
+    // Tracks task count based on command status and swift compiler output
+    private struct TaskCounter {
+        private(set) var finishedCount = 0
+        private(set) var totalCount = 0
+        private var tasks: [Int: String] = [:]
+
+        mutating func commandStatusChanged(_ command: SPMLLBuild.Command, kind: CommandStatusKind) {
             guard command.shouldShowStatus else { return }
 
             switch kind {
             case .isScanning:
-                scanningCount += 1
+                totalCount += 1
             case .isUpToDate:
-                scanningCount -= 1
-                upToDateCount += 1
-                completedCount += 1
+                totalCount -= 1
             case .isComplete:
-                scanningCount -= 1
-                completedCount += 1
+                finishedCount += 1
+            }
+        }
+
+        mutating func swiftCompilerDidOuputMessage(_ message: SwiftCompilerMessage) {
+            switch message.kind {
+            case .began:
+                totalCount += 1
+            case .finished:
+                finishedCount += 1
+            default:
```

Comment:
Avoid using `default` because we can miss future cases (less likely in this case but still).